x-pack/plugins/security_solution/public/common/components/grouping/query/index.ts

```diff
@@ -121,12 +83,13 @@ export const getGroupingQuery = ({
               terms: stackByMultipleFields0.map((stackByMultipleField0) => ({
                 field: stackByMultipleField0,
               })),
+              size: MAX_QUERY_SIZE,
```

Comment:
this is the fix to pagination bug, everything else in this PR is types and tests 👍